### PR TITLE
Feat added Authentication 

### LINKED ISSUE #11: Feat: Adding Auth
Adding Authentication to the App

### PR DESCRIPTION
**Pull request by coldpigli**
Added Login, Signup and Logout feature

### CODE REVIEW

File: src/pages/SignupPage/SignupPage.jsx [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+    firstName: "",
+    lastName: "",
+    email: "",
+    password: "",
+  });
+  const [confirmPass, setConfirmPass] = useState("");
+  const { dispatchUser } = useUserDetails();
+  const navigate = useNavigate();
+  const [errorMsg, setErrorMsg] = useState("");
+  console.log(signupData);
```
Comment (sadathshariff) [inline src/pages/SignupPage/SignupPage.jsx:19]: remove console logs

File: src/pages/SignupPage/SignupPage.jsx [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+                type="text"
+                value={signupData.firstName}
+                placeholder="Enter first name"
+                required
+                autocomplete="on"
+                onChange={(e) => handleChange(e)}
+              />
+            </div>
+            <div className={`input-wrapper gap-d20 ${styles.inputField}`}>
+              <label className="input-name txt-bold" for="input">
```
Comment (sadathshariff) [inline src/pages/SignupPage/SignupPage.jsx]: use htmlFor here as react ignores for

**Review by sadathshariff — APPROVED** [review]
Overall The Code looks good, Suggested some points

File: src/pages/SignupPage/SignupPage.jsx (hunk 2 above)
Comment (coldpigli) [reply src/pages/SignupPage/SignupPage.jsx]: Oh yes. Almost forgot this. Thanks for pointing it out.

File: src/contexts/UserContext/userDetailsReducer.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
+const userDetailsReducer = (state, action) => {
+  switch (action.type) {
+    case "USER_LOGIN":
+      const { firstName, history, likes, playlists } = action.payload;
```
Comment (srejitk) [inline src/contexts/UserContext/userDetailsReducer.js:4]: You can destructure your action to make you code more cleaner.

File: src/pages/SignupPage/SignupPage.jsx [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+        const response = await axios.post("/api/auth/signup", signupData);
+        if (response.status === 200 || response.status === 201) {
+          const { createdUser, encodedToken } = response.data;
+          localStorage.setItem("authToken",encodedToken)
+          dispatchUser({ type: "USER_LOGIN", payload: createdUser });
+          navigate("/explore/All");
+        }
+      } catch (err) {
+        setErrorMsg("Couldn't signup");
+        console.log("Uh!oh. Something went wrong", err);
```
Comment (srejitk) [inline src/pages/SignupPage/SignupPage.jsx:48]: Remove the logs and try implementing a Toast

**Review by srejitk — APPROVED** [review]
Looks good. Just make sure to remove the console logs and replace the for in labels with html for.

Also liked the idea of adding names to the header. Good Work.

**Pull request merged**